### ISSUE TITLE
Move Entity time type check off of Entity

### ISSUE THREAD

**Issue by thehomebrewnerd** [alteryx/featuretools#1369]
The current code for `Entity` has a check to verify that if the Entity has a time index, that the time index is of the **same type as other Entities** in an EntitySet. This check happens during setting of the Entity time index in `Entity.set_time_index`. 

When Entity is replace with a Woodwork dataframe, this validation check will no longer be present and should be implemented in other places where necessary. One likely place where this check could be needed is in `calculate_feature_matrix` but there may be other places as well.

Current validation code included for reference:
```python
        if self.entityset.time_type is None:
            self.entityset.time_type = time_type
        elif self.entityset.time_type != time_type:
            raise TypeError("%s time index is %s type which differs from"
                            " other entityset time indexes" %
                            (self.id, time_type))
```

TLDR: **Time Index should be same type for all Entities in EntitySet**
